apps/extension/src/ui/domains/Sign/Qr/ScanQr.tsx

```diff
@@ -109,30 +110,38 @@ const Scanner = ({
   const inputMenu = useRef(null)
   useClickAway(inputMenu, () => setShowInputMenu(false))
 
+  const codeReader = useAtomValue(codeReaderAtom)
+
   useEffect(() => {
+    if (!codeReader) return
     if (!preview.current) return
 
     const aborted = new AbortController()
-
-    const codeReader = new BrowserQRCodeReader()
-
+    let abortedSignalSet = false
     codeReader
       .decodeFromVideoDevice(selectedVideoInput, preview.current, (result, error, controls) => {
+        if (!abortedSignalSet) {
+          aborted.signal.addEventListener("abort", () => controls.stop())
+          abortedSignalSet = true
+        }
         if (aborted.signal.aborted) return
-        if (error?.name === "NotFoundException") return
 
-        if (error) onError?.(new Error(error.name, { cause: error }))
+        if (error instanceof NotFoundException) return
+        if (error instanceof ChecksumException) return
+        if (error instanceof FormatException) return
```

Comment:
`instanceof` won't break on minified builds, unlike the previous `error?.name ===` test